packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -424,6 +424,12 @@ def __lldb_init_module(debugger: lldb.SBDebugger, _):
   /// This parent folder of `Runner.xcodeproj`.
   @override
   Directory get hostAppRoot {
+    if (parent.manifest.isModule || (parent.manifest.ios?.sharedDarwinSource ?? false)) {
+      final Directory darwinDir = parent.directory.childDirectory('darwin');
```

Comment:
Why was this change needed?

I don't think `hostAppRoot` is the right spot to change, if it is needed for some reason. That's used for apps (or plugin example apps), not for plugins.
I think instead you may want to move the `ios` /`darwin` logic into the  `_editableDirectory` getter, since that's used in more that would otherwise need editing.  For example `_flutterLibRoot` isn't going to point the correct place with this PR.